sonar-ws/src/main/gen-java/org/sonarqube/ws/QualityProfiles.java

```diff
@@ -0,0 +1,2483 @@
+          return this;
+        }
+        /**
+         * <code>optional string languageName = 4;</code>
+         */
+        public Builder setLanguageNameBytes(
+            com.google.protobuf.ByteString value) {
+          if (value == null) {
+    throw new NullPointerException();
+  }
+  bitField0_ |= 0x00000008;
+          languageName_ = value;
+          onChanged();
+          return this;
+        }
+
+        private boolean isInherited_ ;
+        /**
+         * <code>optional bool isInherited = 5;</code>
+         */
+        public boolean hasIsInherited() {
+          return ((bitField0_ & 0x00000010) == 0x00000010);
+        }
+        /**
+         * <code>optional bool isInherited = 5;</code>
+         */
+        public boolean getIsInherited() {
+          return isInherited_;
+        }
+        /**
+         * <code>optional bool isInherited = 5;</code>
+         */
+        public Builder setIsInherited(boolean value) {
+          bitField0_ |= 0x00000010;
+          isInherited_ = value;
+          onChanged();
+          return this;
+        }
+        /**
+         * <code>optional bool isInherited = 5;</code>
+         */
+        public Builder clearIsInherited() {
+          bitField0_ = (bitField0_ & ~0x00000010);
+          isInherited_ = false;
+          onChanged();
+          return this;
+        }
+
+        private java.lang.Object parentKey_ = "";
+        /**
+         * <code>optional string parentKey = 6;</code>
+         */
+        public boolean hasParentKey() {
+          return ((bitField0_ & 0x00000020) == 0x00000020);
+        }
+        /**
+         * <code>optional string parentKey = 6;</code>
+         */
+        public java.lang.String getParentKey() {
+          java.lang.Object ref = parentKey_;
+          if (!(ref instanceof java.lang.String)) {
+            com.google.protobuf.ByteString bs =
+                (com.google.protobuf.ByteString) ref;
+            java.lang.String s = bs.toStringUtf8();
+            if (bs.isValidUtf8()) {
+              parentKey_ = s;
+            }
+            return s;
+          } else {
+            return (java.lang.String) ref;
+          }
+        }
+        /**
+         * <code>optional string parentKey = 6;</code>
+         */
+        public com.google.protobuf.ByteString
+            getParentKeyBytes() {
+          java.lang.Object ref = parentKey_;
+          if (ref instanceof String) {
+            com.google.protobuf.ByteString b = 
+                com.google.protobuf.ByteString.copyFromUtf8(
+                    (java.lang.String) ref);
+            parentKey_ = b;
+            return b;
+          } else {
+            return (com.google.protobuf.ByteString) ref;
+          }
+        }
+        /**
+         * <code>optional string parentKey = 6;</code>
+         */
+        public Builder setParentKey(
+            java.lang.String value) {
+          if (value == null) {
+    throw new NullPointerException();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Throw some other exception here, such as "IllegalArgumentException". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1695)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 13. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)